api/auth.go

```diff
@@ -38,6 +38,11 @@ func withToken(w http.ResponseWriter, r *http.Request) (context.Context, error)
 		return ctx, nil
 	}
 
+	if bearerToken == a.config.OperatorToken {
+		log.Info("Making operator request")
```

Comment:
If the intent is to allow the operator token to behave like an admin, then you probably at least want to call:
`ctx = gcontext.WithAdminFlag(ctx, true)`. I do not know how most endpoints will behave without a valid JWT token, when they expect one.

Reply:
No, we just need to skip JWT validation if the token is the OperatorToken.

When there is an Authorization header gocommerce always try to verify the JWT - and this makes sense so we would give error messages in case of a malformed JWT.

However, the OperatorToken is just a plain Bearer token sent in the Authorization, so without this check all valid requests to the /instances part of the API fails with an "invalid token" error.

Reply:
Yes, but you turned this on for the entire API. I'm worried about the downstream API endpoints that need a set of claims from a JWT (User or Admin).

Reply:
There was already support for the operator token to the `/instances` API endpoints: https://github.com/netlify/gocommerce/blob/5071a6ee8dad88ce5d5451d4c30f20a794f406f3/api/middleware.go#L56

Reply:
The JWT check was turned on for all the API, and happens before the check for the operator token, so all valid requests to `/instances` were failing with a "Token Invalid" error (I ran into this).

This rule just loads claims if there are any, since many methods will work for both ananymous and signed in users. Actions that requires the claims to be there uses `authRequired` or `adminRequired`.

Reply:
This `withToken` should not apply to the `/instances` routes because they are separate routers:
https://github.com/netlify/gocommerce/blob/5071a6ee8dad88ce5d5451d4c30f20a794f406f3/api/api.go#L76

Is that not the case?

Reply:
We actually have automated tests on those endpoints that don't fail with that error.

Reply:
That is strange - all attempts to provision and instance failed with "Invalid Token" from within withToken until I added the check to ignore operator tokens

Reply:
And in any case, it's fine with me that adding a valid operator token in the Authorization header doesn't give invalid token errors.

Again, the withToken check is not there to enforce that there's a valid token - it accepts requests with no Authorization headers, it's just there to process the claims in case the request is signed with a valid JWT.

Reply:
Fair enough. I just thought it was odd you were getting that error.